vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -968,6 +970,28 @@ protected void onAttach(AttachEvent attachEvent) {
         registerClientCloseHandler();
     }
 
+    /**
+     * Sets the ARIA role for the overlay element, used by screen readers.
+     *
+     * @param role
+     *            the role to set
+     */
+    public void setOverlayRole(String role) {
```

Comment:
I'm wondering if this should be an enum on the flow side with both possible values (dialog / alertdialog) to ensure the typical type safety on java side.. those values don't change often and people could theoretically always use the element API if a new value comes out of the blue and you could not update in time.

Reply:
Good point. Maybe we could add an override for this method to Dialog and also Popover.

Reply:
That you have to decide :) While in alpha.. I would just ditch the old methods completely before you have to maintain them for decades

Reply:
We are about to release beta so I'm afraid for now we'd have to leave these methods as they are. 
Created #6645 so that we don't forget about this idea, and maybe consider it for the next minor.